tests/test_astronomer_namespace_pools.py

```diff
@@ -0,0 +1,241 @@
+from tests.helm_template_generator import render_chart
+import pytest
+import yaml
+from . import supported_k8s_versions, get_containers_by_name
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+def test_astronomer_namespace_pools_rbac(kube_version):
+    """Test that helm renders astronomer/commander RBAC resources properly when working with namespace pools"""
+
+    # rbacEnabled and clusterRoles and namespacePools set to true, should create Roles and Rolebindings for namespace in Pool
+    # and ignore the cluster role configuration
+    namespaces = ["my-namespace-1", "my-namespace-2"]
+    docs = render_chart(
+        kube_version=kube_version,
+        values={
+            "global": {
+                "rbacEnabled": True,
+                "clusterRoles": True,
+                "features": {
+                    "namespacePools": {
+                        "enabled": True,
+                        "namespaces": {"create": True, "names": namespaces},
+                    },
+                },
+            }
+        },
+        show_only=[
+            "charts/astronomer/templates/commander/commander-role.yaml",
+            "charts/astronomer/templates/commander/commander-rolebinding.yaml",
+        ],
+    )
+
+    assert len(docs) == 4
+
+    # assertions on Role objects
+    for i in range(0, 2):
+        role = docs[i]
+
```

Comment:
This is more idiomatic:
```suggestion
    for role in docs:
```

Reply:
The thing is I created another list `namespaces` to compare the name of the namespaces to be created (line 45)

Reply:
Ah, yeah, I missed that.